From 4d5166d1833e42d81b854374aa6e73f83a67a70e Mon Sep 17 00:00:00 2001
From: Matthias Clasen <mclasen@redhat.com>
Date: Tue, 27 Nov 2012 23:30:45 +0000
Subject: Fix a crash on 32bit systems

The login-frequency property of the User interface is defined
as uint64 type. Therefore, when setting it with g_object_set,
a 64bit value is collected, and passing an int might lead to
bad things.
https://bugs.freedesktop.org/show_bug.cgi?id=57343
---
diff --git a/src/daemon.c b/src/daemon.c
index ed3a114..4d6f3ab 100644
--- src/daemon.c
+++ src/daemon.c
@@ -207,7 +207,7 @@ daemon_local_user_is_excluded (Daemon *daemon, const gchar *username, const gcha
 #ifdef HAVE_UTMPX_H

 typedef struct {
-        int frequency;
+        guint64 frequency;
         gint64 time;
         GList *previous_logins;
 } UserAccounting;
--
cgit v0.9.0.2-2-gbebe
